packages/utils/src/lib/transform.ts

```diff
@@ -169,3 +170,34 @@ export function toOrdinal(value: number): string {
   return `${value}th`;
 }
 /* eslint-enable no-magic-numbers */
+
+export function tableToFlatArray(table: Table) {
+  const { headings = [], items } = table;
+  const firstItemKeys =
+    typeof items.at(0) === 'string'
+      ? ['Value']
+      : Object.keys(items.at(0) ?? {});
+  const preparedHeadings = headings.flatMap(({ key, label }) => label ?? key);
+  const preparedTHeadersData =
+    preparedHeadings.length > 0 ? preparedHeadings : firstItemKeys;
+
+  const headingKeys = headings.flatMap(({ key }) => key);
+  const itemKeys = headingKeys.length > 0 ? headingKeys : firstItemKeys;
+
+  return [
+    preparedTHeadersData,
+    ...items.filter(Boolean).map(item => {
+      if (typeof item === 'string') {
+        return [item];
+      }
+
+      if (!Array.isArray(item)) {
+        return itemKeys.map(
+          key => (item as Record<typeof key, string>)[key] ?? '',
+        );
+      }
+
+      return [''];
+    }),
+  ] satisfies string[][];
```

Comment:
It's better annotate the function return type than using `return ... satisfies T`, then its part of the function declaration.